text/0101-new-scale-control-model-based-on-IO-size.md

```diff
@@ -0,0 +1,99 @@
+
+```
+config set limit v2 // Switch flow limit v2
+store limit all 500 // All stores can at most move 500MB data per second 
+store limit 1 200   // Store 1 can at most move 200MB data per second
+```
+
+
+### How to adjust windows size 
+The minimum(initial) value of the sliding window is 1000MB. The store can be considered to be scheduled if it's available sending size is bigger than 100MB, so the biggest region can also be scheduled. 
+When the operator execution duration is less than 2 times the snapshot task execution time, the sender windows size will increase , otherwise it will decrease. A PI controller is used to prevent repeated adjustments. It should be noted that when all schedules don't need such big size, it will stop increasing. 
+
+### How to ensure fairness 
+The new scheduler policy does not guarantee that the snapshot sending size of all stores is exactly the same, but will switch to the next candidate region when the store has no available sending size. During the prototype testing process on the scale-out and scale-in scene , the sending size on all TIKV reaches the bottleneck. 
+
+### Scheduler 
+#### Balance Region Scheduler 
+The scheduler picks the store as source first, then picks the region on the source store, finally picks the target store. On the original basis, it is necessary to check the target region leader whether there is enough sending space to send. 
+#### High Priority Scheduler 
+Some scheduler may create a highest priority operator such as balance hot region, region check. For the highest operators (such as hot region operator), they can use this external space if the normal space is not enough. The default external space is about 20% of the send window. This can prevent this window size from being completely occupied by low priority task.
+
+```
+|------- Used -------|
+                     |--------Available -----------|
+|------ Normal ------------|---- Privilege —-------|
+```
+As shown in the 
+
+
+## Compatibility
+### Store Limit
+The store limiter mainly has the following functions that cannot be replaced:
+- The second highest (lowest) score store can still be scheduled. An instance cannot continuously be the source or target to be scheduler because the token is limited in the short term.
+- Remove too many peers from the same store can bring major compaction, which affect the cluster performance.
+### Multi RocksDB
+In this design, the generation of the snapshot will use check-point hard link to replace the scanning of RocksDB. It can reduce the CPU loads and the duration of snapshot generator. But the snapshots are not compressed ,the network and disk bandwidth will increase. 
+The snapshot synchronization process is basically the same, some tasks need to wait to be executed when the store has many snapshot tasks. The waiting time is still positively related to the backlog of the snapshot task, so feedback mechanism still satisfies this situation. 
```

Comment:
When using multi rocksdb, we can make improvement to reduce storage and bandwidth that is impossible when sharing rocksdb. For example, if there is an SST file with the same size, name and hash in the old db, the file can be hard linked directly without actually replication.

Reply:
yes, it will decrease the duration of generating snapshot in multi rocksDB by using hard link directly.